evince.profile: noblacklist bookmark folder

samtinel opened the pull request:
bookmarks are saved unter $HOME/.local/share/gvfs-metadata
since evince is the primary pdf reader, a firejailed evince can't read
or write those
this commit makes them available in the sandbox

glitsj16 commented on etc/profile-a-l/evince.profile:
```diff
@@ -7,6 +7,7 @@ include evince.local
 include globals.local
 
 noblacklist ${HOME}/.config/evince
+noblacklist ${HOME}/.local/share/gvfs-metadata
```
We don't seem to blacklist that path anywhere AFAICT, so this doesn't make much sense. Are you by any chance needing this somehow because of options in a evince.local file?

samtinel replied:
hmm, that line I have added is the only line in my .local for evince. It doesn't work without it.
I will do more digging next week, maybe I have changed some global settings I can't remember.

SkewedZeppelin commented:
> We don't seem to blacklist that path anywhere AFAICT

Wow, we don't.
Crazy!
gvfs-metadata contains information on all storage, static or removable
and also file access history from some programs.

Just run
strings .local/share/gvfs-metadata/*
and you'll see file you recently used.

rusty-snake commented:
Fedora 32; Evince 3.36.7; Firejail 0.9.65;

This is the minimal evince.local I could find which allowed me to view/add/remove bookmarks. gvfs/gio is likely doing the same as dconf/gsettings: read via file-system (performance) and write via D-Bus service (avoid corruptions).

```
ignore dbus-user none

noblacklist ${HOME}/.local/share/gvfs-metadata

dbus-user filter
dbus-user.talk org.gtk.vfs.Daemon
dbus-user.talk org.gtk.vfs.Metadata
```

samtinel commented:
> Fedora 32; Evince 3.36.7; Firejail 0.9.65;
> 
> This is the minimal evince.local I could find
Updated the PR with your changes  - but judging from the first two reactions, I doubt this will be merged?
If that is the case, please close

rusty-snake reviewed (changes requested):
remarks just on the profile ordering:

- Now there are two `noblacklist ${HOME}/.local/share/gvfs-metadata`, remove one.
- The D-Bus stuff should go at the end.

thoughts:
Maybe it would be good to make this opt-in.

samtinel commented:
> remarks just on the profile ordering:
fixed
> thoughts:
> Maybe it would be good to make this opt-in.
This is probably the better idea, I commented out the changes and added an explaining comment - if someone wants it, they can uncomment the lines to use bookmarks.

rusty-snake commented on etc/profile-a-l/evince.profile:
```diff
@@ -55,3 +59,9 @@ private-tmp
 # might break two-page-view on some systems
 dbus-user none
 dbus-system none
+
+# Also uncomment these four lines if you want to use bookmarks
+#ignore dbus-user none
+#dbus-user filter
+#dbus-user.talk org.gtk.vfs.Daemon
+#dbus-user.talk org.gtk.vfs.Metadata
```
A `ignore` after the line which should be ignored did not work.

```suggestion
# might break two-page-view on some systems
dbus-user filter
# Also uncomment (or put in your evince.local) these four lines if you want to use bookmarks
#dbus-user.talk org.gtk.vfs.Daemon
#dbus-user.talk org.gtk.vfs.Metadata
dbus-system none
```

rusty-snake commented on etc/profile-a-l/evince.profile:
```diff
@@ -6,6 +6,10 @@ include evince.local
 include globals.local
 
+# Uncomment this line and the bottom ones to use bookmarks
+# NOTE: This possibly exposes information, including file history from other programs.
```
The allowed D-Bus names below
- allow editing of all metadata and (u)mounting (if I'm correct)
- maybe allow filesystem access outside of the sandbox (=escape)

glitsj16 commented:
@rusty-snake Any progress here? I see a 'Changes requested' message that looks to be resolved.

rusty-snake commented:
What about?
```
# might break two-page-view on some systems
dbus-user filter
# Also uncomment these two lines if you want to use bookmarks
#dbus-user.talk org.gtk.vfs.Daemon
#dbus-user.talk org.gtk.vfs.Metadata
dbus-system none
```

rusty-snake commented:
merged, thanks.

0x29a commented:
@rusty-snake, does this profile still work for you? I've uncommented all specified lines, but can't make evince to store bookmarks.

rusty-snake commented:
I use evince now as flatpak (to have document portal).

Does it work if you add `dbus-user.talk org.gtk.vfs.*`?

0x29a commented:
>Does it work if you add dbus-user.talk org.gtk.vfs.*?

@rusty-snake, nope, same.
